examples/rust/tcp-push-pop.rs

```diff
@@ -28,34 +29,36 @@ pub const SOCK_STREAM: i32 = libc::SOCK_STREAM;
 // Constants
 //======================================================================================================================
 
-const BUFSIZE_BYTES: usize = 64;
-const FILL_CHAR: u8 = 0x65;
-const ITERATIONS: usize = 10;
+const BUF_SIZE_BYTES: usize = 64;
+const ITERATIONS: usize = u8::MAX as usize;
 const TIMEOUT_SECONDS: Duration = Duration::from_secs(256);
 
-fn mksga(libos: &mut LibOS, size: usize, value: u8) -> Result<demi_sgarray_t> {
-    let sga: demi_sgarray_t = match libos.sgaalloc(size) {
+fn mksga(libos: &mut LibOS, value: u8) -> Result<demi_sgarray_t> {
+    let sga: demi_sgarray_t = match libos.sgaalloc(BUF_SIZE_BYTES) {
         Ok(sga) => sga,
         Err(e) => anyhow::bail!("failed to allocate scatter-gather array: {:?}", e),
     };
 
+    // Create pointer for filling the array.
+
+    let ptr: *mut u8 = sga.sga_segs[0].sgaseg_buf as *mut u8;
     // Ensure that allocated array has the requested size.
-    if sga.sga_segs[0].sgaseg_len as usize != size {
+    if sga.sga_segs[0].sgaseg_len as usize != BUF_SIZE_BYTES || ptr.is_null() || !ptr.is_aligned() {
```

Comment:
If any of these conditions are met, it possibly represents sga corruption. Would this signify another serious underlying problem? Or could it just be a faulty send from the other side?

Reply:
I don't think this is a faulty send since this is a new allocated sga array. I took out the pointer alignment check since it must be aligned to bytes.